src/Symfony/Component/Form/Tests/Extension/Core/Type/DateTypeTest.php

```diff
@@ -1156,6 +1156,36 @@ public function testDateTimeImmutableInputTimezoneNotMatchingModelTimezone()
         ]);
     }
 
+    public function testSubmitWithCustomCalendarOption()
+    {
+        $form = $this->factory->create(static::TESTED_TYPE, options: [
+            'format' => 'y-MM-dd',
+            'html5' => false,
+            'input' => 'array',
+            'calendar' => \IntlCalendar::createInstance(locale: 'zh_TW@calendar=roc'),
```

Comment:
IMO the use of this calendar deserves some kind of comment. I believe it's not so obvious when reading the test why we submit a year like 113 but expect 2024 as the final result.

Reply:
I added a comment. Is it clearer to you now?